pkg/cmd/login/login.go

```diff
@@ -321,3 +325,28 @@ func convertToHostname(serverAddress string) (string, error) {
 
 	return serverAddress, nil
 }
+
+func infiniteAuthorizationLoop(responses []*http.Response) bool {
+	n := len(responses)
+	if n == 1 || (n > 1 && !sameUnmetAuthorizationChallenge(responses[n-2], responses[n-1])) {
+		return false
+	}
+
+	return true
+}
```

Comment:
When `n = 0` you return `true`. Is it normal ?
If `n = 0` return false => no need for next iteration to make a propre Authorization

Reply:
It's not, thanks for spotting this. I copied the logic from https://github.com/containerd/containerd/blob/f0ce2f6d369a35d9b0bf3e6a31404ee3dbfdb9bb/remotes/docker/authorizer.go#L332-L344, but did not notice the difference.